pkg/kcl_plugin/hello_plugin/api.go

```diff
@@ -0,0 +1,153 @@
+
+package hello_plugin
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"kusionstack.io/kclvm-go/pkg/kcl_plugin"
+)
+
+func init() {
+	kcl_plugin.RegisterPlugin(kcl_plugin.Plugin{
+		Name:      _Name,
+		MethodMap: _MethodMap,
+		ResetFunc: reset_plugin,
+	})
+}
+
+const _Name = "hello"
+
+var _MethodMap = map[string]kcl_plugin.PluginMethod{
+	// func set_global_int(v int64)
+	"set_global_int": func(args_json, kwargs_json string) (interface{}, error) {
+		args, err := parseArgs(args_json)
+		if err != nil {
+			return nil, err
+		}
+
+		set_global_int(argInt64(args[0]))
+		return nil, nil
+	},
+
+	// func get_global_int() int64
+	"get_global_int": func(args_json, kwargs_json string) (interface{}, error) {
+		v := get_global_int()
+		return float64(v), nil
+	},
+
+	// func say_hello(msg string)
+	"say_hello": func(args_json, kwargs_json string) (interface{}, error) {
+		args, err := parseArgs(args_json)
+		if err != nil {
+			return nil, err
+		}
+
+		msg := fmt.Sprint(args[0])
+		say_hello(msg)
+
+		return nil, nil
+	},
+
+	// func add(a, b int64) int64
+	"add": func(args_json, kwargs_json string) (interface{}, error) {
+		args, err := parseArgs(args_json)
+		if err != nil {
+			return nil, err
+		}
+
+		a := argInt64(args[0])
+		b := argInt64(args[1])
+		v := add(a, b)
+
+		return float64(v), nil
+	},
+
+	// func tolower(s string) string
+	"tolower": func(args_json, kwargs_json string) (interface{}, error) {
+		args, err := parseArgs(args_json)
+		if err != nil {
+			return nil, err
+		}
+
+		s := fmt.Sprint(args[0])
+		v := tolower(s)
+
+		return v, nil
+	},
+
+	// func update_dict(d map[string]interface{}, key, value string) map[string]interface{}
+	"update_dict": func(args_json, kwargs_json string) (interface{}, error) {
+		args, err := parseArgs(args_json)
+		if err != nil {
+			return nil, err
+		}
+
+		d := args[0].(map[string]interface{})
+		key := fmt.Sprint(args[1])
+		value := fmt.Sprint(args[2])
+		v := update_dict(d, key, value)
+
+		return v, nil
+	},
+
+	// func list_append(list []interface{}, values ...interface{}) []interface{}
+	"list_append": func(args_json, kwargs_json string) (interface{}, error) {
+		args, err := parseArgs(args_json)
+		if err != nil {
+			return nil, err
+		}
+
+		d := args[0].([]interface{})
+		v := list_append(d, args[1:]...)
+
+		return v, nil
+	},
+
+	// func foo(a, b interface{}, x []interface{}, values map[string]interface{}) interface{} {
+	"foo": func(args_json, kwargs_json string) (interface{}, error) {
+		args, err := parseArgs(args_json)
+		if err != nil {
+			return nil, err
+		}
+		values, err := parseKwargs(kwargs_json)
+		if err != nil {
+			return nil, err
+		}
+
+		a := args[0]
+		b := args[1]
+		x := args[2:]
+
+		v := foo(a, b, nil, x, values)
+
+		return v, nil
+	},
+}
+
+func argInt64(arg interface{}) int64 {
+	s := fmt.Sprint(arg)
+	v, _ := strconv.ParseInt(s, 10, 64)
+	return v
+}
+
+func parseArgs(args_json string) (args []interface{}, err error) {
```

Comment:
`parseArgs` and `parseKwargs` should be called inside the kclvm-go-plugin framework and not exposed to the user?

Reply:
> `parseArgs` and `parseKwargs` should be called inside the kclvm-go-plugin framework and not exposed to the user?

use KCL Value?

Reply:
> `parseArgs` and `parseKwargs` should be called inside the kclvm-go-plugin framework and not exposed to the user?

remove the parseArgs and parseKwargs, add kcl_plugin.MethodArgs for args, and kcl_plugin.MethodResult for result